dbt/adapters/redshift/connections.py

```diff
@@ -212,20 +212,23 @@ def get_connect_method(self):
             "region": self.credentials.region,
             "timeout": self.credentials.connect_timeout,
         }
+
+        # Use region if possible but otherwise treat it as optional
         if kwargs["region"] is None:
             logger.debug("No region provided, attempting to determine from host.")
             try:
-                region_value = self.credentials.host.split(".")[2]
+                inferred_region = self.credentials.host.split(".")[2]
+                if _is_valid_region(inferred_region):
+                    kwargs["region"] = inferred_region
+                else:
+                    raise IndexError
             except IndexError:
-                raise dbt.exceptions.FailedToConnectError(
-                    "No region provided and unable to determine region from host: "
-                    "{}".format(self.credentials.host)
+                logger.debug(
+                    f"Unable to determine region from host, proceeding without region: {self.credentials.host}"
                 )
 
-            kwargs["region"] = region_value
-
-        # Validate the set region
-        if not _is_valid_region(kwargs["region"]):
+        # Validate the set region if it's available
+        if not kwargs["region"] and not _is_valid_region(kwargs["region"]):
             raise dbt.exceptions.FailedToConnectError(
                 "Invalid region provided: {}".format(kwargs["region"])
             )
```

Comment:
@colin-rogers-dbt I might be reading the logic incorrectly here, but is it supposed to be this?
```python
# Validate the set region if it's available
if kwargs["region"] and not _is_valid_region(kwargs["region"]):
```